src/mongocxx/exception/private/error_category.cpp

```diff
@@ -382,7 +383,12 @@ class mongocxx_error_category_impl final : public std::error_category {
     }
 
     std::string message(int code) const noexcept override {
-        return "unknown mongocxx error";
+        switch (static_cast<error_code>(code)) {
+        case error_code::k_database_uninitialized:
+            return "database is uninitialized";
```

Comment:
That is good, with a small quibble: you can use one, just not for much. How about "invalid use of default-constructed mongocxx::database object"?. That still isn't perfect because it doesn't cover the moved-from case, but it is close enough, I think. Saying "invalid use of default constructed or moved-from mongocxx::database object" seems awfully wordy. What do you think?